auth_api_key/models/auth_api_key.py

```diff
@@ -3,36 +3,64 @@
 
 from odoo import api, models, tools, _
 
-from odoo.addons.server_environment import serv_config
 from odoo.tools import consteq
 
 from odoo.exceptions import ValidationError, AccessError
 
 
-class AuthApiKey(models.TransientModel):
+class AuthApiKey(models.Model):
     _name = "auth.api.key"
+    _inherit = "server.env.mixin"
     _description = "API Key Retriever"
 
+    name = fields.Char(required=True)
+    key = fields.Char(required=True)
+    user = fields.Char(required=True)
+
+    _sql_constraints = [
+        ('key_uniq', 'unique(key)', 'API Key Retriever must be unique !'),
+    ]
+
+    @property
+    def _server_env_fields(self):
+        base_fields = super()._server_env_fields
+        api_key_fields = {
+            "key": {},
+            "user": {},
+        }
+        api_key_fields.update(base_fields)
+        return api_key_fields
+
     @api.model
     @tools.ormcache("api_key")
```

Comment:
The result of this method can't be put into ormcache since it's not allowed to put a recordset into the cache....